torch_geometric/nn/aggr/attention.py

```diff
@@ -45,13 +45,21 @@ def reset_parameters(self):
         reset(self.gate_nn)
         reset(self.nn)
 
-    def forward(self, x: Tensor, index: Optional[Tensor] = None,
-                ptr: Optional[Tensor] = None, dim_size: Optional[int] = None,
-                dim: int = -2) -> Tensor:
+    def forward(
+        self,
+        x: Tensor,
+        index: Optional[Tensor] = None,
+        ptr: Optional[Tensor] = None,
+        dim_size: Optional[int] = None,
+        dim: int = -2,
+        batch: Optional[Tensor] = None,
```

Comment:
@rusty1s This allows the batch and batch_size arguments to be passed to the underlying MLP modules within the AttentionalAggregation. Since we provide the arguments to the MLP class itself, we must provide that same capability to the modules that use the MLP.

Reply:
I don't think we need these new arguments here. Actually, `batch` is already given through `index` here, and `batch_size` is given through `dim_size`. You should be able to just pass these instead.

Reply:
Fixed.